pkg/bloombuild/planner/queue/queue.go

```diff
@@ -71,13 +102,37 @@ func NewQueue(logger log.Logger, cfg Config, limits Limits, metrics *Metrics) (*
 }
 
 func (q *Queue) starting(ctx context.Context) error {
+	// TODO(salvacorts): We can recover the queue from the disk here.
+	// Until then, we just remove all the files in the directory so the disk
+	// doesn't grow indefinitely.
+	if q.cfg.StoreTasksOnDisk && q.cfg.CleanTasksDirectory {
```

Comment:
If we cannot recover tasks from disk yet, we should not expose any settings for it.

`CleanTaskDirectory` should be `false` instead

Reply:
> we should not expose any settings for it.

I don't think we are. Which one do you mean?

`CleanTasksDirectory` will be gone when we are able to recover tasks from the disk.

Until then, if we do not recover from the disk, we will keep leaking files into the disk across restarts.

Reply:
Sorry, ofc it should be `true`. No need for a setting then

Reply:
I mean this setting https://github.com/grafana/loki/pull/14874/files#diff-2bb8e70d0a56554bbc6c93993f0457a2a010af29154d72b12d3492ebf4a7ba46R1236-R1238